src/ast/ddl.rs

```diff
@@ -2157,6 +2157,10 @@ impl fmt::Display for ClusteredBy {
 #[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
 #[cfg_attr(feature = "visitor", derive(Visit, VisitMut))]
 pub struct CreateFunction {
+    /// Conditionally alters the function only if it already exists.
+    ///
+    /// [SQL Server](https://learn.microsoft.com/en-us/sql/t-sql/statements/create-function-transact-sql?view=sql-server-ver16#or-alter)
+    pub or_alter: bool,
     pub or_replace: bool,
```

Comment:
I wasn't sure if there was a conventional ordering for these struct fields so I put it on top, before "or_replace". However I just noticed that parse_create has or_replace above or_alter here: https://github.com/apache/datafusion-sqlparser-rs/blob/d090ad4ccfd6c76c87de676fae50bbaec36e752a/src/parser/mod.rs#L4536-L4537

Happy to adjust it here for consistency if anyone prefers